homeassistant/components/cover/ryobi_gdo.py

```diff
@@ -0,0 +1,117 @@
+"""
+Ryobi platform for the cover component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/cover.ryobi_gdo/
+"""
+from datetime import timedelta
+
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.cover import (
+    CoverDevice, PLATFORM_SCHEMA, SUPPORT_OPEN, SUPPORT_CLOSE)
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD, STATE_UNKNOWN, STATE_CLOSED)
+
+REQUIREMENTS = ['py_ryobi_gdo==0.0.10']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DEVICE_ID = 'device_id'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_DEVICE_ID): [cv.string],
+})
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+SUPPORTED_FEATURES = (SUPPORT_OPEN | SUPPORT_CLOSE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ryobi covers."""
+    from py_ryobi_gdo import RyobiGDO as ryobi_door
+    covers = []
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    devices = config.get(CONF_DEVICE_ID)
+    auth_ok = False
+    device_to_add = False
+    _LOGGER.debug("%s devices to set up", len(devices))
+
+    for device_id in devices:
+        my_door = ryobi_door(username, password, device_id)
+        if my_door.get_api_key() is True:
```

Comment:
Check if it's `False` and return. This way you can eliminate `auth_ok` as there is no need to proceed if the credentials are wrong.

Guard clauses make the code more readable because there is less indent and nesting.

Reply:
done